wordpress-mobile/WordPress-Android
# Zendesk: Update Zendesk plan and addons for the selected site

This PR sets Zendesk highest plan and addons for the selected site that we now get directly from `/me/sites` endpoint - (added in D59613-code), [instead of the custom logic that was used earlier](https://github.com/wordpress-mobile/WordPress-Android/pull/12140/files#diff-523dffdbc9bf63b0dc10a024899c735f2db64cc1a0c6c4f2abef91194f31f491).

Corresponding FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2034

### Review Instructions

While working on this PR, noticed that

1.  we added Zendesk plan and addons for sites in the `/me/sites` API but when switching the site from the site selector, `/sites/[id] `endpoint (where the fields were not set for Jetpack sites), overwrote these fields in the local db setting them to null. To fix this, submitted the D63091-code that sets these fields on the Jetpack sites for `/sites/[id] `endpoint.

     We can either wait for the fix to be deployed, or test it on the sandbox.

2. `selectedSite` was `null` on the `MeFragment` due to the way it was set. Fixed this behavior in cbccba6, see commit comment for details.

### To test

Prerequisite: 

wpcom account with a site having at least one add-on plan:

```
jetpack_addon_backup_daily
jetpack_addon_backup_realtime
jetpack_addon_search
jetpack_addon_scan_daily
jetpack_addon_scan_realtime
jetpack_addon_antispam
```

1. Add a breakpoint at the below lines:

https://github.com/wordpress-mobile/WordPress-Android/blob/2fe33b667437353ba3535987b16a07f66bff7b3b/WordPress/src/main/java/org/wordpress/android/support/ZendeskHelper.kt#L397

https://github.com/wordpress-mobile/WordPress-Android/blob/47cafffa7bc55f69c88b009d17596340b44b0665/WordPress/src/main/java/org/wordpress/android/support/ZendeskHelper.kt#L467

2. Login using the account mentioned in prerequisites, and switch to the site with add-ons on My Sites tab.

3. Go to My Site -> Me Profile -> Help & Support -> Contact Support.

4. Notice that the Zendesk plan and addons are correctly set for the selected site at the breakpoints. 

5. Submit the ticket.

6. Notice that the plan and add-ons are correctly set in the Zendesk form.

### Merge Instructions

FluxC 

1. Wait for https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2034 to be reviewed and merged
2. Create a tag in FluxC and update it in `build.gradle`

PS: Add-Ons are set as tags and not as a field due to findings here: p4a5px-2CT-p2#comment-12093. Currently, SLAs are not set based on these add-ons. The behavior might change during SLA implementation. The only change I foresee in this case is sending the Zendesk addons as Zendesk addons field value instead of sending them as tags. We might want to wait merging this PR till that time or merge it now and update it later on if needed. 

## Regression Notes
1. Potential unintended areas of impact 🟢


2. What I did to test those areas of impact (or what existing automated tests I relied on)  🟢


3. What automated tests I added (or what prevented me from doing so)  🟢

PR submission checklist:
